### ISSUE TITLE
Using inference_demo.py with cme_head but get poor result on the video in demo page

### ISSUE THREAD

**Issue by TMM233** [ClaudiaCuttano/SAMWISE#29]
Hello, thank you for this amazing work.
However, when I tried to replicate the code, I couldn't find the correct mask like on the demo page.
My inference_demo.py uses the ‘pretrain/final_model_mevis.pth’ checkpoint and also use cme_head.

<img width="1305" height="625" alt="Image" src="https://github.com/user-attachments/assets/9ba35593-a159-46db-843c-096b0dcc7345" />
Below are the less than ideal results I got when using the prompt from the demo.

https://github.com/user-attachments/assets/5e042d1f-3b8a-4f75-98ef-75c01c730adf
A_woman_holding_a_plastic_bag,_taking_photos_with_her_other_hand

https://github.com/user-attachments/assets/2b1ac448-af00-48c6-9d04-3af8ea8734a1

The_individual_dismounting_from_the_motorcycle_and_standing_next_to_it

https://github.com/user-attachments/assets/c9a647fb-5040-4543-9ab2-aa2bcc8e825f

I am certain that my prompt and mask video correspond to each other.
Do you have any suggestions?

Additionally, I would like to be able to perform spatial relative queries on my own data.
For example, I want to query for two cups without using color information:
“The cup, the one placed on the gray book”
“The cup, the one directly placed on the desk”
Are there any shortcomings with these queries?

<img width="588" height="333" alt="Image" src="https://github.com/user-attachments/assets/26bcb974-3671-40b4-b0d4-d92df7fc4970" />

**Comment by ga1i13o** [comment]
Hi @TMM233,

Thanks for your interest and sorry about the delay!

Regarding the demo, we generated those qualitative results using a different checkpoint from the one currently released. We’ve noticed that the produced masks are not always perfectly consistent across runs, even though the overall performance remains fairly stable.

As for your question about spatial relative queries, the current model is not well-suited for those cases. The training data does not include that type of relational phrasing. RefCOCO mainly focuses on categorical descriptions and absolute spatial terms like “left” or “right” within the image, while MEViS is more oriented toward action-based descriptions. Queries such as “the cup on the gray book” or “the cup on the desk” involve compositional spatial reasoning that the model hasn’t explicitly seen during training, so performance there is likely to be unreliable. If you have access to such data it's probably best to finetune the model on it.

Hope this helps clarify things, and thanks again for taking the time to explore SAMWISE!

Best,
Gabriele
